File: CloudFormation/src/main/java/com/intellij/aws/cloudformation/CloudFormationBundle.java
package com.intellij.aws.cloudformation;

import com.intellij.CommonBundle;
import org.jetbrains.annotations.PropertyKey;

import java.util.ResourceBundle;

public class CloudFormationBundle {
  private static final ResourceBundle ourBundle = ResourceBundle.getBundle(CloudFormationBundle.class.getName());

  private CloudFormationBundle() {
  }

  public static String getString(@PropertyKey(resourceBundle = "com.intellij.aws.cloudformation.CloudFormationBundle") String key,
                                 Object... params) {
    return CommonBundle.message(ourBundle, key, params);
  }
}

File: CloudFormation/src/main/kotlin/com/intellij/aws/cloudformation/CloudFormationBundle.kt
package com.intellij.aws.cloudformation

import com.intellij.DynamicBundle
import org.jetbrains.annotations.Nls
import org.jetbrains.annotations.NonNls
import org.jetbrains.annotations.PropertyKey

@NonNls
private const val BUNDLE = "messages.CloudFormationBundle"

object CloudFormationBundle {
  private val instance = DynamicBundle(CloudFormationBundle::class.java, BUNDLE)

  @Nls
  fun message(@PropertyKey(resourceBundle = BUNDLE) key: String, vararg params: Any): String {
    return if (instance.containsKey(key)) instance.getMessage(key, *params) else CloudFormationDeprecatedMessagesBundle.message(key, *params)
  }

  @JvmStatic
  fun messagePointer(@PropertyKey(resourceBundle = BUNDLE) key: String,
                     vararg params: Any): java.util.function.Supplier<String> {
    return if (instance.containsKey(key))  instance.getLazyMessage(key, *params) else CloudFormationDeprecatedMessagesBundle.messagePointer(key, *params)
  }
}